test/components/views/rooms/RoomHeader-test.tsx

```diff
@@ -0,0 +1,252 @@
+        expect(initial.innerHTML).toEqual("Z");
+
+        // And there is no image avatar (because it's not set on this room)
+        const image = findImg(rendered, ".mx_BaseAvatar_image");
+        expect(image.src).toEqual("data:image/png;base64,00");
+    });
+
+    test('shows the room avatar in a DM with only ourselves', () => {
+        // When we render a non-DM room with 1 person in it
+        const room = createRoom({ name: "Z Room", isDm: true, users: [] });
+        const rendered = render(room);
+
+        // Then the room's avatar is the initial of its name
+        const initial = findSpan(rendered, ".mx_BaseAvatar_initial");
+        expect(initial.innerHTML).toEqual("Z");
+
+        // And there is no image avatar (because it's not set on this room)
+        const image = findImg(rendered, ".mx_BaseAvatar_image");
+        expect(image.src).toEqual("data:image/png;base64,00");
+    });
+
+    test('shows the user avatar in a DM with 2 people', () => {
+        // Note: this is the interesting case - this is the ONLY
+        //       time we should use the user's avatar.
+
+        // When we render a DM room with only 2 people in it
+        const room = createRoom(
+            { name: "Y Room", isDm: true, users: ["other"] });
+        const rendered = render(room);
+
+        // Then we use the other user's avatar as our room's image avatar
+        const image = findImg(rendered, ".mx_BaseAvatar_image");
+        expect(image.src).toEqual(
+            "http://this.is.a.url/example.org/other");
+
+        // And there is no initial avatar
+        expect(
+            rendered.querySelectorAll(".mx_BaseAvatar_initial"),
+        ).toHaveLength(0);
+    });
+
+    test('shows the room avatar in a DM with >2 people', () => {
+        // When we render a DM room with 3 people in it
+        const room = createRoom({
+            name: "Z Room", isDm: true, users: ["other1", "other2"] });
+        const rendered = render(room);
+
+        // Then the room's avatar is the initial of its name
+        const initial = findSpan(rendered, ".mx_BaseAvatar_initial");
+        expect(initial.innerHTML).toEqual("Z");
+
+        // And there is no image avatar (because it's not set on this room)
+        const image = findImg(rendered, ".mx_BaseAvatar_image");
+        expect(image.src).toEqual("data:image/png;base64,00");
+    });
+});
+
+interface IRoomCreationInfo {
+    name: string;
+    isDm: boolean;
+    users: string[];
+}
+
+function createRoom(info: IRoomCreationInfo) {
+    TestUtils.stubClient();
+    const client: MatrixClient = MatrixClientPeg.get();
+
+    const roomId = '!1234567890:domain';
+    const userId = client.getUserId();
+    if (info.isDm) {
+        client.getAccountData = (eventType) => {
+            expect(eventType).toEqual("m.direct");
+            return mkDirectEvent(roomId, userId, info.users);
+        };
+    }
+
+    DMRoomMap.makeShared().start();
```

Comment:
It doesn't look like you ever call `stop()`. Should it be called at the end of the test?

Reply:
What `stop()` does is call `removeListener` on the client. Since we're using a test client created in `test/test-utils.js`, and `removeListener` does nothing, I think we can get away with not calling `stop()`.

What do you think?

Reply:
Great, let's not worry about that then 👍